Version 0.7-rc1

roihershberg opened the pull request:
Changelog from v0.6:
* A/B slots checking mechanism
* Integration of RootTools and RootShell libraries
* Root detection
* Activity lifecycle handling (for snappy screen rotation and running information commands only at the launch of the app)
* Bottom layout margin

Changelog in README:
* Credits section

roihershberg committed:
Better reboot

It is better to use svc power to reboot through the power manager instead of forcing reboot. I think it prevents corruption of data. If it fails - it will force reboot like before.

roihershberg committed:
Add checking for bootctl utility

This checking for bootctl utility prevents the crash of the app when the utility is not available like in MagiskSU (without the module that fixes that).
It shows a dialog that informs the user about that.

roihershberg committed:
Change to version 0.6

Changing the version to v0.6

roihershberg committed:
Add screenshot file of v0.6

Screenshot of v0.6

roihershberg committed:
Update README.md

Added instructions for MagiskSU and updated the screenshot to v0.6

roihershberg committed:
Remove unnecessary imports

Removed unnecessary imports.

roihershberg committed:
Merge branch 'master' of https://github.com/roihershberg/Switch-My-Slot-Android

roihershberg committed:
Merge remote-tracking branch 'upstream/master'

roihershberg committed:
Changed the reading of the buffer

The readLine function with DataInputStream is depracted - switching to BufferedReader.
Also removing the call to dismiss the error dialog because the dialog is already dismissed.

roihershberg committed:
Merge remote-tracking branch 'upstream/master'

roihershberg committed:
Update README.md

Removed ARM64 requirement.

roihershberg committed:
Fix detection of A/B device

Fixed the detection of A/B device and added displaying of error dialog if android version is lower than Nougat 7.0

roihershberg committed:
Update README.md

Added requirement for android Nougat 7.0 or newer.

roihershberg committed:
Correcting minimum API level

Minimum android version with seamless A/B system updates according to Wikipedia is 7.1 and not 7.0

roihershberg committed:
Update .gitignore

roihershberg committed:
Merge remote-tracking branch 'upstream/master'

roihershberg committed:
Merging V0.7 to master (#1)

* Add RootTools library

Added RootTools library to the project.

* Add RootShell library and exitApp function

Added the RootShell library which is the core of RootTools and moved the exit mechanism to a dedicated function.

* Change the code structure of A/B checking

Changed the code structure of the A/B checking and added support for checking Virtual A/B.

* Fix crashing when bootctl is missing

Fixed the crash when bootctl utility is missing.

* Root detection and integration of RootTools library

Added integration of RootTools and RootShell libraries - replaced all su executions with the libraries.
Added root detection dialog (if the device isn't rooted) and root denied dialog.

* Bottom layout margin

Added 10dp bottom margin in the layout.

* Changed version to v0.7

* Fixed get-suffix command

Moved the slot suffix command to the current slot command so it is created when the current slot is known.

* Update screenshot

Updated screenshot to v0.7

* Update README.md

Added credits.

* Activity lifecycle handling

Better handling of activity lifecycle. Added preservation of UI information and shell when activity get recreated. This allows snappy screen rotation and executing information commands only once.

* Fixed ViewModel integration

Assigning the shell to the ViewModel after it got created.

* Seperation of getting root shell

Seperation of getting a root shell to a dedicated function.

* Fixed back button exit

Fixed the bug that when the user presses the back button and opens the app again, there's a blank screen. The disadvantage with this fix is that the app immediately close with no closing animation.

* Move ABResult to ABChecker

Moved the ABResult to be inside ABChecker.

* Add bottom margin in landscape

Added the bottom margin also in the landscape layout.

* Add comments to some functions

roihershberg committed:
Fix checkUtil method (#2)

* Add RootTools library

Added RootTools library to the project.

* Add RootShell library and exitApp function

Added the RootShell library which is the core of RootTools and moved the exit mechanism to a dedicated function.

* Change the code structure of A/B checking

Changed the code structure of the A/B checking and added support for checking Virtual A/B.

* Fix crashing when bootctl is missing

Fixed the crash when bootctl utility is missing.

* Root detection and integration of RootTools library

Added integration of RootTools and RootShell libraries - replaced all su executions with the libraries.
Added root detection dialog (if the device isn't rooted) and root denied dialog.

* Bottom layout margin

Added 10dp bottom margin in the layout.

* Changed version to v0.7

* Fixed get-suffix command

Moved the slot suffix command to the current slot command so it is created when the current slot is known.

* Update screenshot

Updated screenshot to v0.7

* Update README.md

Added credits.

* Activity lifecycle handling

Better handling of activity lifecycle. Added preservation of UI information and shell when activity get recreated. This allows snappy screen rotation and executing information commands only once.

* Fixed ViewModel integration

Assigning the shell to the ViewModel after it got created.

* Seperation of getting root shell

Seperation of getting a root shell to a dedicated function.

* Fixed back button exit

Fixed the bug that when the user presses the back button and opens the app again, there's a blank screen. The disadvantage with this fix is that the app immediately close with no closing animation.

* Move ABResult to ABChecker

Moved the ABResult to be inside ABChecker.

* Add bottom margin in landscape

Added the bottom margin also in the landscape layout.

* Add comments to some functions

* Modified checkUtil in RootTools library

Modified checkUtil in RootTools to accept also 750 permission.

roihershberg committed:
Update settings.gradle

Removed duplication of include

roihershberg committed:
Update MainActivity.java

Updated a comment.

roihershberg committed:
Added constants for missing commands

Added constants for switch slot and reboot commands.

roihershberg committed:
Update comment

Updating the exitApp comment

roihershberg committed:
Update MainActivity.java

Change SDK to SDK_INT instead of parsing the string.

roihershberg committed:
Fixed hanging shells

It now closing the shells but there are still hanging threads so I added calling to System.exit function.

roihershberg commented:
Wow, I found a library that is so much better from RootTools and RootShell. It is from the creator of Magisk and it is constantly maintained. It is called [libsu](https://github.com/topjohnwu/libsu) and it's so easy to use. You can merge this PR but wait with the release of 0.7. I will probably replace the current libraries with libsu and create a PR for 0.7-rc2.

gibcheesepuffs commented:
I tested it but it results in a dialog saying bootctl is missing.

roihershberg commented:
Did you pulled the latest commits?

roihershberg commented:
I can include RootTools and RootShell libraries as source code instead of an aar file. We can then modify the libraries directly from this project. What do you prefer?

gibcheesepuffs commented:
Yes, I believe that I have pulled the latest commits. Also, I think adding the source code approach is better.

roihershberg commented:
Ok. So I will change it from aar to source and add a new method to check that a utility is exists and has executable permissions. When I'm done I will send here another comment.

roihershberg committed:
Change integration of libraries

Changed the integration of RootTools and RootShell libraries from pre-compiled to source code.

roihershberg committed:
Add checkUtilExecutable method

Added checkUtilExecutable to RootTools library and changed the call in MainActivity from checkUtil to the new method.

roihershberg commented:
I pushed the changes.

roihershberg committed:
Add new lines at the end of files

roihershberg committed:
Move closing of BufferedReader

Moved the closing of BufferedReader to before the return of the function.

roihershberg committed:
Remove su retries

Removed su retires when the access is denied.

gibcheesepuffs commented:
Reviewed, tested, merged, and released! 🎉 
Thanks once again for your contribution.
